cyipopt/scipy_interface.py

```diff
@@ -61,19 +61,32 @@ class IpoptProblemWrapper(object):
         Epsilon used in finite differences.
 
     """
-    def __init__(self, fun, args=(), kwargs=None, jac=None, hess=None,
-                 hessp=None, constraints=(), eps=1e-8):
+    def __init__(self,
+                 fun,
+                 n,
+                 args=(),
+                 kwargs=None,
+                 jac=None,
+                 hess=None,
+                 hessp=None,
+                 constraints=(),
+                 con_dims=(),
+                 eps=1e-8):
```

Comment:
Adding the arg `n` is technically backward incompatible since you can supply kwargs in order as args. Can `n` be a kwarg added at the end? Same goes for the order of `con_dims`.

Reply:
> Can `n` be a kwarg added at the end? Same goes for the order of `con_dims`.

Of course. Just to be sure I don't confuse it with the first `kwargs`: You mean something like

``` python
def __init__(self, fun, args=(), kwargs=None, jac=None, hess=None, hessp=None, constraints=(), eps=1e-8, **kkwargs):
    self.n = kkwargs.get('n', None)
    # ....
    self.con_dims = kkwargs.get('con_dims', [])
    # ....
```
right?

Reply:
Can this work?

```python
def __init__(self, fun, args=(), kwargs=None, jac=None, hess=None, hessp=None, constraints=(), eps=1e-8, n=None:
```